### ISSUE TITLE
Change the definition of the RXX gate

### ISSUE THREAD

**Issue by Cryoris** [Qiskit/qiskit-terra#3857]
### Summary

When computing gradients of a circuit, the cost is proportional to the number of parameterized gates in the circuit. The current implementation of `RXXGate(theta)` uses the parameter `theta` in three gates in its definition, however we can reduce this to one.
This other implementation changes the global phase of the RXX, but: the existing implementation also contains a global phase.

Therefore we could change the 3-parameter up-to-global-phase implementation to a 1-parameter up-to-global-phase implementation.

### Details

Comparison of the two implementations:
```
(U3Gate(np.pi / 2, theta, 0), [q[0]], []),  # change to (HGate(), [q[0]], []),
(HGate(), [q[1]], []),
(CnotGate(), [q[0], q[1]], []),
(U1Gate(-theta), [q[1]], []),  # change -theta to theta
(CnotGate(), [q[0], q[1]], []),
(HGate(), [q[1]], []),
(U2Gate(-np.pi, np.pi - theta), [q[0]], []), # change to (HGate(), [q[0]], []),
```
The current implementation has a global phase of `exp(iθ/2)` while the proposed one has one of `exp(-iθ/2)`.

**Comment by chriseclectic** [comment]
I verified in Mathematica and new def looks good to me. I would make one additional change:

* Swap `U1Gate(theta)` with `RzGate(theta)`

Then the final matrix would be correct including phase if the Rz matrix was used (ignoring Rz unrolling to U1 with incorrect phase which will be fixed eventually)

**Comment by Cryoris** [comment]
Agreed, also it makes the circuit diagram more intuitive seeing `Rz` instead of `U1`. I'll open a PR.

**Comment by chriseclectic** [comment]
In the same PR you could also change the def of the RZZ gate to use Rz instead of U1 for the same reason. it should be:
```
[(CnotGate(), [q[0], q[1]], []),
 (RzGate(self.params[0]), [q[1]], []),
 (CnotGate(), [q[0], q[1]], [])]
```

**Comment by Cryoris** [comment]
Closed by #3913.
